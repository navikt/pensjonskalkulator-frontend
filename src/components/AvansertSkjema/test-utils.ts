export function getPreviousMonth(): string {
  const today = new Date()
  const currentMonth = parseInt(
    today.toLocaleDateString('en-CA').slice(5, 7),
    10
  ) // Extracts the month from the date string
  const previousMonth = currentMonth - 1
  return previousMonth.toString()
}
