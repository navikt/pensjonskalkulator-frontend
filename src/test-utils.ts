import renderer from 'react-test-renderer'

import { render } from '@testing-library/react'

export function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => children,
    ...options,
  })

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
// override render export
export { customRender as render }
