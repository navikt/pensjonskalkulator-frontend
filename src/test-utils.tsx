import React, { PropsWithChildren } from 'react'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'

import { PreloadedState, createListenerMiddleware } from '@reduxjs/toolkit'
import { render, RenderOptions } from '@testing-library/react'

import { createSamtykkeListener } from './state/listeners/samtykkeListener'
import {
  setupStore,
  RootState,
  AppStore,
  AppStartListening,
} from './state/store'
import { getTranslation_nb } from './translations/nb'
export interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export const swallowErrors = (testFn: () => void) => {
  const error = console.error
  console.error = () => {}
  testFn()
  console.error = error
}

export const swallowErrorsAsync = async (testFn: () => Promise<void>) => {
  const cache = console.error
  console.error = () => {}
  await testFn()
  console.error = cache
}

function generateMockedTranslations() {
  const nbTranslations = getTranslation_nb()
  const translations: Record<string, string> = {}
  for (const key in nbTranslations) {
    translations[key] = key
  }
  return translations
}

// Return an object with the store and all of RTL's query functions
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState, true),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
    return (
      <Provider store={store}>
        <IntlProvider locale={'nb'} messages={generateMockedTranslations()}>
          {children}
        </IntlProvider>
      </Provider>
    )
  }
  const listenerMiddleware = createListenerMiddleware()
  createSamtykkeListener(listenerMiddleware.startListening as AppStartListening)

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'

export { renderWithProviders as render }
