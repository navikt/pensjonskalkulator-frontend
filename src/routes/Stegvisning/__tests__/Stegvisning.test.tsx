import { describe, it } from 'vitest'

import { Stegvisning } from '..'
import { render, screen } from '@/test-utils'

describe('Stegvisning', () => {
  it('rendrer slik den skal, med riktig heading', () => {
    const result = render(<Stegvisning />)

    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent(
      'Midlertidig stegvisning'
    )
    expect(result.asFragment()).toMatchSnapshot()
  })
})
