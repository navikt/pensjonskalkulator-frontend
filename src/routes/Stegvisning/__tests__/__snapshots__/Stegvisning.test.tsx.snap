// Vitest Snapshot v1

exports[`Stegvisning > rendrer slik den skal, med riktig heading 1`] = `
<DocumentFragment>
  <section
    class="_section_d75732"
  >
    <h2>
      Midlertidig stegvisning
    </h2>
  </section>
</DocumentFragment>
`;
